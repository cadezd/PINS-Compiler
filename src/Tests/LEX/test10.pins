# primer malce bolj kompleksnega testa, ki bo preveril vecino stvari
integer:indeks =        0;
logical:strazar = false;
string:niz = 'danes je res lep ''dan'' - uporabili smo enojne narekovaje ! ! !';

# pa pričnimo
while(indeks > -1000){
    indeks = indeks-1;
    if (indeks <= -500) then
        indeks = indeks * 1;
    } else {
        indeks = indeks / 10;
    }
    for (var i = 0; i != 100; i = i + 1) { # zanka zanka zanka
        i = niz[i] % 123;
    }
    if (i == 50 | i == 59 & i > 40) {
        i, besedilo = 0;
    }
}
# se vec narekovajev
'tu jih je pa res veliko '''''', kar trije se ponovijo -_q231453%$#!%& aaaa'
imej se lepo se naprej! # bye bye