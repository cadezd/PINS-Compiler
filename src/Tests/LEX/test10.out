[2:1-2:7] AT_INTEGER:integer
[2:8] OP_COLON::
[2:9-2:14] IDENTIFIER:indeks
[2:16] OP_ASSIGN:=
[2:25] C_INTEGER:0
[2:26] OP_SEMICOLON:;
[3:1-3:7] AT_LOGICAL:logical
[3:8] OP_COLON::
[3:9-3:15] IDENTIFIER:strazar
[3:17] OP_ASSIGN:=
[3:19-3:23] C_LOGICAL:false
[3:24] OP_SEMICOLON:;
[4:1-4:6] AT_STRING:string
[4:7] OP_COLON::
[4:8-4:10] IDENTIFIER:niz
[4:12] OP_ASSIGN:=
[4:14-4:79] C_STRING:danes je res lep 'dan' - uporabili smo enojne narekovaje ! ! !
[4:80] OP_SEMICOLON:;
[7:1-7:5] KW_WHILE:while
[7:6] OP_LPARENT:(
[7:7-7:12] IDENTIFIER:indeks
[7:14] OP_GT:>
[7:16] OP_SUB:-
[7:17-7:20] C_INTEGER:1000
[7:21] OP_RPARENT:)
[7:22] OP_LBRACE:{
[8:5-8:10] IDENTIFIER:indeks
[8:12] OP_ASSIGN:=
[8:14-8:19] IDENTIFIER:indeks
[8:20] OP_SUB:-
[8:21] C_INTEGER:1
[8:22] OP_SEMICOLON:;
[9:5-9:6] KW_IF:if
[9:8] OP_LPARENT:(
[9:9-9:14] IDENTIFIER:indeks
[9:16-9:17] OP_LEQ:<=
[9:19] OP_SUB:-
[9:20-9:22] C_INTEGER:500
[9:23] OP_RPARENT:)
[9:25-9:28] KW_THEN:then
[10:9-10:14] IDENTIFIER:indeks
[10:16] OP_ASSIGN:=
[10:18-10:23] IDENTIFIER:indeks
[10:25] OP_MUL:*
[10:27] C_INTEGER:1
[10:28] OP_SEMICOLON:;
[11:5] OP_RBRACE:}
[11:7-11:10] KW_ELSE:else
[11:12] OP_LBRACE:{
[12:9-12:14] IDENTIFIER:indeks
[12:16] OP_ASSIGN:=
[12:18-12:23] IDENTIFIER:indeks
[12:25] OP_DIV:/
[12:27-12:28] C_INTEGER:10
[12:29] OP_SEMICOLON:;
[13:5] OP_RBRACE:}
[14:5-14:7] KW_FOR:for
[14:9] OP_LPARENT:(
[14:10-14:12] KW_VAR:var
[14:14] IDENTIFIER:i
[14:16] OP_ASSIGN:=
[14:18] C_INTEGER:0
[14:19] OP_SEMICOLON:;
[14:21] IDENTIFIER:i
[14:23-14:24] OP_NEQ:!=
[14:26-14:28] C_INTEGER:100
[14:29] OP_SEMICOLON:;
[14:31] IDENTIFIER:i
[14:33] OP_ASSIGN:=
[14:35] IDENTIFIER:i
[14:37] OP_ADD:+
[14:39] C_INTEGER:1
[14:40] OP_RPARENT:)
[14:42] OP_LBRACE:{
[15:9] IDENTIFIER:i
[15:11] OP_ASSIGN:=
[15:13-15:15] IDENTIFIER:niz
[15:16] OP_LBRACKET:[
[15:17] IDENTIFIER:i
[15:18] OP_RBRACKET:]
[15:20] OP_MOD:%
[15:22-15:24] C_INTEGER:123
[15:25] OP_SEMICOLON:;
[16:5] OP_RBRACE:}
[17:5-17:6] KW_IF:if
[17:8] OP_LPARENT:(
[17:9] IDENTIFIER:i
[17:11-17:12] OP_EQ:==
[17:14-17:15] C_INTEGER:50
[17:17] OP_OR:|
[17:19] IDENTIFIER:i
[17:21-17:22] OP_EQ:==
[17:24-17:25] C_INTEGER:59
[17:27] OP_AND:&
[17:29] IDENTIFIER:i
[17:31] OP_GT:>
[17:33-17:34] C_INTEGER:40
[17:35] OP_RPARENT:)
[17:37] OP_LBRACE:{
[18:9] IDENTIFIER:i
[18:10] OP_COMMA:,
[18:12-18:19] IDENTIFIER:besedilo
[18:21] OP_ASSIGN:=
[18:23] C_INTEGER:0
[18:24] OP_SEMICOLON:;
[19:5] OP_RBRACE:}
[20:1] OP_RBRACE:}
[22:1-22:76] C_STRING:tu jih je pa res veliko ''', kar trije se ponovijo -_q231453%$#!%& aaaa
[23:1-23:4] IDENTIFIER:imej
[23:6-23:7] IDENTIFIER:se
[23:9-23:12] IDENTIFIER:lepo
[23:14-23:15] IDENTIFIER:se
[23:17-23:22] IDENTIFIER:naprej
[23:23] OP_NOT:!
EOF:EOF